tests/test_pretest.py

```diff
@@ -373,16 +374,47 @@ def test_collect_pfc_pause_delay_params(duthosts, tbinfo):
     except IOError as e:
         logger.warning('Unable to create file {}: {}'.format(filepath, e))
 
-
-def test_update_saithrift_ptf(request, ptfhost):
+def test_update_saithrift_ptf(request, ptfhost, duthosts, enum_dut_hostname):
     '''
     Install the correct python saithrift package on the ptf
     '''
     py_saithrift_url = request.config.getoption("--py_saithrift_url")
     if not py_saithrift_url:
         pytest.skip("No URL specified for python saithrift package")
+
+    duthost = duthosts[enum_dut_hostname]
+    output = duthost.shell("show version", module_ignore_errors=True)['stdout']
+    version_reg = re.compile(r"sonic software version: +([^\s]+)\s", re.IGNORECASE)
+    asic_reg = re.compile(r"asic: +([^\s]+)\s", re.IGNORECASE)
+    version = version_reg.findall(output)[0] if version_reg.search(output) else "" # sample value: SONiC.20240510.33, SONiC.20250505.07, SONiC.internal.129741107-8524154c2d, SONiC.master.882522-695c23859
+    asic = asic_reg.findall(output)[0] if asic_reg.search(output) else "" # only broadcom, cisco-8000, mellanox support qos sai tests
+
+    if "master" in version:
+        branch_name = "master"
+    elif "internal" in version:
+        branch_name = "internal"
+    else:
+        # Extract year/month from version string to determine branch
+        date_match = re.search(r'(\d{4})(\d{2})', version)
+        if date_match:
+            year, month = date_match.groups()
+            branch_name = f"internal-{year}{month}"
+        else:
+            pytest.fail("Unable to parse or recognize version format: {}".format(version))
+
+    debian_codename = "bookworm"
```

Comment:
If debian is upgraded, we still need to change this hardcode, and code sync is much slower than environment update.